compiler/tests/evm_runtime.rs

```diff
@@ -187,6 +187,22 @@ fn test_runtime_sloadn() {
     })
 }
 
+#[test]
+fn test_weird_bug() {
+    with_executor(&|mut executor| {
+        test_runtime_functions(
+            &mut executor,
+            functions::std(),
+            statements! {
+                (let min_val := 0x8000000000000000000000000000000000000000000000000000000000000000)
+                (let val1 := 0x8000000000000000000000000000000000000000000000000000000000000000)
+                (let val2 := 0x01)
+                [assert_eq!(0, (and((and((slt(val1, 0)), (sgt(val2, 0)))), (slt(val1, (sdiv(min_val, val2)))))))]
```

Comment:
So this is returning true and we expect false? Could be a solidity bug, I don't quite understand the logic here.

Reply:
Yes, this is the exact line (from the entire check) that causes us to revert. The actual check is [here](https://github.com/ethereum/fe/pull/271#discussion_r583583219)

Computing `i256_min_value * 1` in solidity with remix works just fine. And when I take a snippet of solidity that has that code and transpile it to yul with `solc snippet.sol --ir` then this check is what I get. And this is how I created all the checks so far and all of them except this one worked.

One thing that I could think of is that there's a bug with this check in the generated YUL that I get from `solc` and that it works in remix because it doesn't go through the YUL step and actually uses a different computation than the one that I get from `solc snippet.sol --ir`. I'll give it another shot today.